core/src/main/java/org/apache/carbondata/core/util/CarbonUtil.java

```diff
@@ -2741,12 +2741,28 @@ public static void copyCarbonDataFileToCarbonStorePath(String localFilePath,
     try {
       CarbonFile localCarbonFile =
           FileFactory.getCarbonFile(localFilePath, FileFactory.getFileType(localFilePath));
+      if (localCarbonFile.getSize() == 0L) {
+        LOGGER.error("The file size of local carbon file: " + localFilePath + " is 0.");
+        throw new CarbonDataWriterException(
+            "The file size of local carbon file is 0.");
+      }
       String carbonFilePath = carbonDataDirectoryPath + localFilePath
           .substring(localFilePath.lastIndexOf(File.separator));
       copyLocalFileToCarbonStore(carbonFilePath, localFilePath,
           CarbonCommonConstants.BYTEBUFFER_SIZE,
           getMaxOfBlockAndFileSize(fileSizeInBytes, localCarbonFile.getSize()));
-    } catch (IOException e) {
+      CarbonFile targetCarbonFile =
+          FileFactory.getCarbonFile(carbonFilePath, FileFactory.getFileType(carbonFilePath));
```

Comment:
```suggestion
          FileFactory.getCarbonFile(carbonFilePath);
```
and move to previous line